packages/flutter/lib/src/rendering/object.dart

```diff
@@ -2228,12 +2277,15 @@ abstract class RenderObject extends AbstractNode with DiagnosticableTreeMixin im
     assert(!owner!._debugDoingPaint);
     assert(isRepaintBoundary);
     assert(_layer != null); // use scheduleInitialPaint the first time
-    _layer!.detach();
+    _layer!..detach()..dispose();
     _layer = rootLayer;
     markNeedsPaint();
   }
 
+  DisposalHandle? _disposalHandle;
```

Comment:
From API design review: Consider changing `Layer` to hold the picture.

However, we also have to cover a case where many pictures are painted by a single render object, e.g. it paints to a picture, paints children that push layer(s), and paints to a picture again.